react-native-spokestack.d.ts

```diff
@@ -0,0 +1,123 @@
+export = RNSpokestack;
+export as namespace RNSpokestack;
+
+declare namespace RNSpokestack {
+  function constructor(props?: object): void;
+
+  enum TraceLevel {
+    DEBUG = 10,
+    PERF = 20,
+    INFO = 30,
+    NONE = 100,
+  }
+
+  enum TTSFormat {
+    TEXT = 0,
+    SSML = 1,
+    SPEECHMARKDOWN = 2,
+  }
+
+  interface SpokestackRecognizeEvent {
+    transcript?: string;
+  }
+
+  interface SpokestackErrorEvent {
+    error?: string;
+  }
+
+  interface SpokestackTraceEvent {
+    message?: string;
+  }
+
+  interface SpokestackTTSEvent {
+    url?: string;
+  }
+
+  interface NLUSlot {
+    type: string;
+    value: string;
+  }
+
+  interface SpokestackNLUEvent {
+    result?: {
+      intent: string;
+      confidence: string;
+      slots: NLUSlot[];
+    };
+  }
+
+  interface SynthesizeOptions {
+    input: string;
+    format: TTSFormat;
+    voice: string;
+  }
+
+  type SpokestackEvent = SpokestackRecognizeEvent &
+    SpokestackErrorEvent &
+    SpokestackTraceEvent &
+    SpokestackTTSEvent &
+    SpokestackNLUEvent;
+
+  interface SpokestackConfig {
+    input: string;
+    stages: string[];
+    properties?: {
+      "agc-compression-gain-db"?: number;
+      "agc-target-level-dbfs"?: number;
+      "frame-width"?: number;
+      "fft-window-size"?: number;
+      "fft-hop-length"?: number;
+      locale?: string;
+      "pre-emphasis"?: number;
+      "sample-rate"?: number;
+      "vad-fall-delay"?: number;
+
+      // Wakeword
+      wakewords?: string;
+      "wake-filter-path"?: string;
+      "wake-detect-path"?: string;
+      "wake-encode-path"?: string;
+      "wake-phrase-length"?: number;
+      "wake-smooth-length"?: number;
+      "wake-threshold"?: number;
+      "wake-active-min"?: number;
+      "wake-active-max"?: number;
+
+      "google-credentials"?: string; // only set if using `GoogleSpeechRecognizer` stage above
```

Comment:
for azure asr support, there are additional config properties: https://javadoc.io/doc/io.spokestack/spokestack-android/latest/io/spokestack/spokestack/microsoft/AzureSpeechRecognizer.html